packages/react-components/src/text-input/src/TextInput.jsx

```diff
@@ -152,6 +152,7 @@ export function InnerTextInput(props) {
                     className: cssModule(
                         "o-ui-input",
                         iconMarkup && "has-icon",
+                        disabled && "disabled",
```

Comment:
You don't need to add this class, the CSS selector can target the native "disabled" attribute.

Reply:
on the input element yes but I need a way to be aware that the whole input is disabled in order to add a not allowed cursor to child like an icon, this is not possible by targeting the input element only.